src/filter.jl

```diff
@@ -58,19 +58,28 @@ function filterODE(du, u, p, t)
   return nothing
 end
 
-function backwardfilter(k::SDEKernel, p::WGaussian{(:μ, :Σ, :c)}; alg=Euler(), inplace=false)
-  message, solend = backwardfilter(k::SDEKernel, NamedTuple{(:logscale, :μ, :Σ)}((p.c, p.μ, p.Σ)); alg=alg, inplace=inplace)
+function backwardfilter(k::SDEKernel, p::WGaussian{(:μ, :Σ, :c)}; alg=Euler(),
+    inplace=false, apply_timechange=false)
+  message, solend = backwardfilter(k::SDEKernel, NamedTuple{(:logscale, :μ, :Σ)}((p.c, p.μ, p.Σ));
+    alg=alg, inplace=inplace, apply_timechange=apply_timechange)
   return message, WGaussian{(:μ, :Σ, :c)}(myunpack(solend)...)
 end
 
-function backwardfilter(k::SDEKernel, (c, ν, P)::NamedTuple{(:logscale, :μ, :Σ)}; alg=Euler(), inplace=false)
+function backwardfilter(k::SDEKernel, (c, ν, P)::NamedTuple{(:logscale, :μ, :Σ)};
+    alg=Euler(), inplace=false, apply_timechange=false)
   @unpack trange, p = k
 
   # Initialize OD
   u0 = mypack(ν, P, c)
 
   prob = ODEProblem{inplace}(filterODE, u0, reverse(get_tspan(trange)), p)
-  sol = solve(prob, alg, dt = get_dt(trange))
-  message = Message(sol, k)
+  if !apply_timechange
+    sol = solve(prob, alg, dt = get_dt(trange))
```

Comment:
This should only be called if trange is a range and not a vector? Perhaps three cases: apply_timechange, isa AbstractRange, else

Reply:
Yeah, I did this probably too coarse grained.. I also didn't think deeply enough how to treat the adaptive solvers yet. Does the following make sense?
-  non-adaptive, no time change applied (`apply_timechange=false`): use `tstops = trange` (which should work for both `trange::AbstractRange` and `trange::AbstractVector `. Presumably this could be better than using `dt`, since `range` corrects for floating point errors. This argument would actually apply to all `solve` calls..)
- non-adaptive,  time change applied `apply_timechange=true`: use `tstops=timechange(trange)` 
- adaptive, `apply_timechange=nothing`: use initial `dt` from `trange` and no `tstops`  [Is this one needed? I guess it might be good to allow an adaptive solver to step however the error estimate drives it..]
- adaptive, `apply_timechange=true`:  use `tstops=timechange(trange)`, and set initial `dt` as above.
- adaptive, `apply_timechange=false`  use `tstops=trange`, and set initial `dt` as above.

Reply:
tricky, we have a keyword argument which changes the type of a quantity from range to vector, could be that this requires a function barrier.

Reply:
isn't the use of an extra function `timechange(trange)` already a function barrier?